lib/stringify.js

```diff
@@ -101,7 +101,7 @@ const BSON_TO_JS_STRING = {
       o += 'm';
     }
 
-    return `RegExp('${v.source}'${hasOptions ? `, ${o}` : ''})`;
+    return `RegExp('${v.source}'${hasOptions ? `, '${o}'` : ''})`;
```

Comment:
This is tangential, but since you’re here – can you check what happens if `v.source` contains a single quote?

Generally speaking, for generating something that will be parsed as a JS string from an actual JS string object, I would always use `JSON.stringify()`, as much as I prefer single quotes:

```suggestion
    return `RegExp(${JSON.stringify(v.source)}'${hasOptions ? `, ${JSON.stringify(o)}` : ''})`;
```

Reply:
Apparently it just works, somehow:

```javascript
      it('escapes quotes', function() {
        var res = parser.parseFilter(
          '{name: /\'/}'
        );
        var stringified = parser.stringify(res);
        assert.equal(
          stringified,
          "{name: RegExp('\'')}"
        );
      });
```

But I suppose it won't hurt to add that.

Reply:
that’s surprising, but good – I guess you can leave this as-is then :)